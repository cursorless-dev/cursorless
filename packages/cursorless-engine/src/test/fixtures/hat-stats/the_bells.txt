The Bells
by Edgar Allan Poe 1809 – 1849 (public domain)

I.

        Hear the sledges with the bells—
                 Silver bells!
What a world of merriment their melody foretells!
        How they tinkle, tinkle, tinkle,
           In the icy air of night!
        While the stars that oversprinkle
        All the heavens, seem to twinkle
           With a crystalline delight;
         Keeping time, time, time,
         In a sort of Runic rhyme,
To the tintinabulation that so musically wells
       From the bells, bells, bells, bells,
               Bells, bells, bells—
  From the jingling and the tinkling of the bells.

II.

        Hear the mellow wedding bells,
                 Golden bells!
What a world of happiness their harmony foretells!
        Through the balmy air of night
        How they ring out their delight!
           From the molten-golden notes,
               And all in tune,
           What a liquid ditty floats
    To the turtle-dove that listens, while she gloats
               On the moon!
         Oh, from out the sounding cells,
What a gush of euphony voluminously wells!
               How it swells!
               How it dwells
           On the Future! how it tells
           Of the rapture that impels
         To the swinging and the ringing
           Of the bells, bells, bells,
         Of the bells, bells, bells, bells,
               Bells, bells, bells—
  To the rhyming and the chiming of the bells!

III.

         Hear the loud alarum bells—
                 Brazen bells!
What tale of terror, now, their turbulency tells!
       In the startled ear of night
       How they scream out their affright!
         Too much horrified to speak,
         They can only shriek, shriek,
                  Out of tune,
In a clamorous appealing to the mercy of the fire,
In a mad expostulation with the deaf and frantic fire,
            Leaping higher, higher, higher,
            With a desperate desire,
         And a resolute endeavor
         Now—now to sit or never,
       By the side of the pale-faced moon.
            Oh, the bells, bells, bells!
            What a tale their terror tells
                  Of Despair!
       How they clang, and clash, and roar!
       What a horror they outpour
On the bosom of the palpitating air!
       Yet the ear it fully knows,
            By the twanging,
            And the clanging,
         How the danger ebbs and flows;
       Yet the ear distinctly tells,
            In the jangling,
            And the wrangling.
       How the danger sinks and swells,
By the sinking or the swelling in the anger of the bells—
             Of the bells—
     Of the bells, bells, bells, bells,
            Bells, bells, bells—
 In the clamor and the clangor of the bells!

IV.

          Hear the tolling of the bells—
                 Iron bells!
What a world of solemn thought their monody compels!
        In the silence of the night,
        How we shiver with affright
  At the melancholy menace of their tone!
        For every sound that floats
        From the rust within their throats
                 Is a groan.
        And the people—ah, the people—
       They that dwell up in the steeple,
                 All alone,
        And who tolling, tolling, tolling,
          In that muffled monotone,
         Feel a glory in so rolling
          On the human heart a stone—
     They are neither man nor woman—
     They are neither brute nor human—
              They are Ghouls:
        And their king it is who tolls;
        And he rolls, rolls, rolls,
                    Rolls
             A pæan from the bells!
          And his merry bosom swells
             With the pæan of the bells!
          And he dances, and he yells;
          Keeping time, time, time,
          In a sort of Runic rhyme,
             To the pæan of the bells—
               Of the bells:
          Keeping time, time, time,
          In a sort of Runic rhyme,
            To the throbbing of the bells—
          Of the bells, bells, bells—
            To the sobbing of the bells;
          Keeping time, time, time,
            As he knells, knells, knells,
          In a happy Runic rhyme,
            To the rolling of the bells—
          Of the bells, bells, bells—
            To the tolling of the bells,
      Of the bells, bells, bells, bells—
              Bells, bells, bells—
  To the moaning and the groaning of the bells.
