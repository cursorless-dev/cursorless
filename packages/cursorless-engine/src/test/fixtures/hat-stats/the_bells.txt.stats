nTokens: 782

tokenSpark:
	▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▃▁▁▃▁▃▁▃▁▃▃▁▁▁▁▃▃▃▃▃▁▁▁▃▃▃▃▃▁▃▃▁▃▃▃▃▃▃▃▃▃▃▃▃
	▃▃▃▃▃▁▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▅▃▃▃▃▁▃▃▃▃▃▃▃▃▃▃▃▃▃▃
	▅▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▁▃▃▅▃▃▃▃▅▃▃▁▃▃▃▃▃▃▃▃▃▅▃▃▃▃▃▃▃▃▅
	▃▃▃▃▃▅▃▅▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▃▅▃▃▃▃▃▅▃▅▃▃▃▃▃▃▅▃▅▃▅▃▅▃▅▃▅▃▅▃▅▃▅▃▅▃
	▅▃▃▅▅▃▃▅▃▅▅▃▃▃▃▃▅▃▃▃▃▁▃▃▅▅▅▃▅▃▅▃▃▃▃▃▅▃▃▅▃▅▃▃▅▃▃▅▅▃▃▅▃▅▃▃▃▃▅▃
	▅▅▅▃▅▃▅▃▃▅▅▃▅▅▅▅▃▅▃▁▅▅▃▃▃▅▅▃▃▅▃▅▃▅▅▅▃▃▅▃▅▅▃▅▃▅▅▅▅▃▅▃▅▅▅▅▃▃▃▃
	▃▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▃▅▅▃▃▅▅▃▅▅▅▅▅▅▅▃▃▅▅▃▅▅▃▅▅▃▅▅▅▃▃▅▃▅▃▅▅▅▃█▅▅▃
	▅▅▅▃▃▅▅▃▅█▅▅▃█▅▅▃▃▅▅▃▃▅▅█▃▅▃▅▅▃▅▅▃▅▅▅▃▅▅▅▃▅▅▅█▅█▅█▅█▅█▅█▅▃▅▅
	▃▅▅▅▅▅▅▅▃▃▅▅▅▅▅▅▃▅▅▅▅▅▅▅▃▅▅▃▃▅▅▅▅▅▅▅█▅▅▃▅▅▅▅▃▅▅▅▅▅▅▃▅▅▅▅▅▅▅▅
	▅▅▅▅▃▅▅▃▃▅█▅▃▃▃▅▅▃▅▅▃█▅▅█▅▅▅█▅█▅█▅▅▅▅█▅▅▅▅▅▅▅▅▅▅▅▅▃▅▅▅▅▅▅▅▅▅
	▅▅▅▅▅▅▅▅▁▅▅▃▅▅▅▅▃▅▅▅█▅█▅█▅▅▃▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅▅█▅▅▅▃▃▅█▅█▅█▅
	▅▅▅▅▅█▅▅▅▅▅▅▅▅▅▅▃▃▅█▅█▅█▅█▅▅▅▅█▅▅▅▅▅▅▅▅▅▅█▅█▅▅█▅▅▅▅▅▃▃▅█▅█▅█
	▅▅▃█▅█▅█▅█▅▅▅██▅▅▅▅▅▅▅▅▅█▅█▅▅█▅▅▅▅▅█▅▅▅█▅█▅█▅▅▅█▅█▅▅█▅▅▅▅▅▅▅
	▅▃

nHats:
	mean: 93%
	std: 0%
	min: 93%
	max: 94%
	spark: 0              25               50               75         █     100

nPenalty0:
	mean: 5%
	std: 0%
	min: 5%
	max: 5%
	spark: 0   █          25               50               75               100

nPenalty1:
	mean: 40%
	std: 0%
	min: 40%
	max: 40%
	spark: 0              25        █▃     50               75               100

nPenalty2:
	mean: 49%
	std: 0%
	min: 48%
	max: 49%
	spark: 0              25             █ 50               75               100
