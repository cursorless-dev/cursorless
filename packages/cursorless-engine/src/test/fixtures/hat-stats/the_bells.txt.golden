_   _
The Bells
[-] [---]

 _ _     _     _   _    _  _   _ _     _     _
by Edgar Allan Poe 1809 – 1849 (public domain)
[] [---] [---] [-] [--] | [--] |[----] [----]|

__
I.
||

        _    A   _       _    B     _  A
        Hear the sledges with the bells—
        [--] [-] [-----] [--] [-] [---]|

                    _   A    _
                 Silver bells!
                 [----] [---]|

A    A  _     _ _             _ A      A        A
What a world of merriment their melody foretells!
[--] | [---] [] [-------] [---] [----] [-------]|

          B C       _  _   _   A    A  B
        How they tinkle, tinkle, tinkle,
        [-] [--] [----]| [----]| [----]|

           A  D    _  B    B   _  B
           In the icy air of night!
           [] [-] [-] [-] [] [---]|

        C     E   A     F     A
        While the stars that oversprinkle
        [---] [-] [---] [--] [----------]

        C   1   A      C B    2  3
        All the heavens, seem to twinkle
        [-] [-] [-----]| [--] [] [-----]

           D    D A           A      _
           With a crystalline delight;
           [--] | [---------] [-----]|

         B       4   D 5   E 6   F
         Keeping time, time, time,
         [-----] [--]| [--]| [--]|

         B  E C    A   A    A    1
         In a sort of Runic rhyme,
         [] | [--] [] [---] [---]|

 B  B          B        C   D   C        E
To the tintinabulation that so musically wells
[] [-] [-------------] [--] [] [-------] [---]

       C     D  B    2 C    3 D    4 E    5
       From the bells, bells, bells, bells,
       [--] [-] [---]| [---]| [---]| [---]|

                           A
               F    6 1    1 2    B
               Bells, bells, bells—
               [---]| [---]| [---]|

  D     E  _        F    F   C       C   1  3    A
  From the jingling and the tinkling of the bells.
  [--] [-] [------] [-] [-] [------] [] [-] [---]|

D B
II.
[]|

                                     A
        2     3  B      F       4    2
        Hear the mellow wedding bells,
        [--] [-] [----] [-----] [---]|

                 A      5    C
                 Golden bells!
                 [----] [---]|

1    1 2     D  4          5    6       E        D
What a world of happiness their harmony foretells!
[--] | [---] [] [-------] [---] [-----] [-------]|

            D     A 6     2   E  A
        Through the balmy air of night
        [-----] [-] [---] [-] [] [---]

         F    B  B     E    C   B      E
        How they ring out their delight!
        [-] [--] [--] [-] [---] [-----]|

                                       A
           F      D C     _B      B    3
           From the molten-golden notes,
           [--] [-] [----]|[----] [---]|

                              A
               3   4   E   F  4
               And all in tune,
               [-] [-] [] [--]|

           3    5   _    C     1
           What a liquid ditty floats
           [--] | [----] [---] [----]

                                   A
     1   E  1    AD      6  A      5 4     E   C
    To the turtle-dove that listens, while she gloats
    [] [-] [----]|[--] [--] [-----]| [---] [-] [----]

               2    F D   F
               On the moon!
               [] [-] [--]|

           A                            B
         3 6 2     2    1   3      B    1
         Oh, from out the sounding cells,
         []| [--] [-] [-] [------] [---]|

     A
5    1  4   4   5         6         6    1
What a gush of euphony voluminously wells!
[--] | [--] [] [-----] [----------] [---]|

                5  F  F     2
               How it swells!
               [-] [] [----]|

                6  1  E
               How it dwells
               [-] [] [----]

                          A
            C   2 3     3 1   2   3
           On the Future! how it tells
           [] [-] [----]| [-] [] [---]

                          A
            4   4 C       1    3
           Of the rapture that impels
           [] [-] [-----] [--] [----]

         A
         2    5 1         D    6 D
         To the swinging and the ringing
         [] [-] [------] [-] [-] [-----]

              A        B      B      B
            5 3     B  2   C  3   D  4
           Of the bells, bells, bells,
           [] [-] [---]| [---]| [---]|

            A        B      B      C      C
          6 4     E  5   F  6   1  1   2  2
         Of the bells, bells, bells, bells,
         [] [-] [---]| [---]| [---]| [---]|

                    C      C
                 3  3   4  4   5  C
               Bells, bells, bells—
               [---]| [---]| [---]|

  A  A               B           A  B
  5  6   E        E  1   C       1  2     6  4
  To the rhyming and the chiming of the bells!
  [] [-] [-----] [-] [-] [-----] [] [-] [---]|

4  C
III.
[-]|

              B
            F 3      F    1       2D
         Hear the loud alarum bells—
         [--] [-] [--] [----] [---]|

                    _       35
                 Brazen bells!
                 [----] [---]|

A    B    A        C    C
1    4    2    2   5 F  6    5    3            46
What tale of terror, now, their turbulency tells!
[--] [--] [] [----]| [-]| [---] [--------] [---]|

          B                A
       6  5   5          4 3  1
       In the startled ear of night
       [] [-] [------] [-] [] [---]

       A                A                A
       2      A 6       1      5    6    1
       How they scream out their affright!
       [-] [--] [----] [-] [---] [------]|

         B                  C       D
         6   E            1 1   A   1
         Too much horrified to speak,
         [-] [--] [-------] [] [---]|

                             D       D
            B D    2        C2      D3
         They can only shriek, shriek,
         [--] [-] [--] [----]| [----]|

                   A  A      D
                   2  4    3 4
                  Out of tune,
                  [-] [] [--]|

   A                     C  C         A  C   A   D
 4 2 E          B        2  3   F     5  4   1   5
In a clamorous appealing to the mercy of the fire,
[] | [-------] [-------] [] [-] [---] [] [-] [--]|

   A                   A    C                    A   D
 5 3 1    _            2    5   2     6        F 2   6
In a mad expostulation with the deaf and frantic fire,
[] | [-] [-----------] [--] [-] [--] [-] [-----] [--]|

                          E       E       E
               C      D   1   E   2   F   3
            Leaping higher, higher, higher,
            [-----] [----]| [----]| [----]|

            A    A                 E
            3    4 3         4     4
            With a desperate desire,
            [--] | [-------] [----]|

             A      A
           5 5      3     6
         And a resolute endeavor
         [-] | [------] [------]

         A   A   C  A   A       E
         1  E2   6  1   6    B  5
         Now—now to sit or never,
         [-]|[-] [] [-] [] [---]|

          D   A    B  D
        C 1   2    1  2   D   B  1   2   D
       By the side of the pale-faced moon.
       [] [-] [--] [] [-] [--]|[---] [--]|

            B E D   A    F A    F A    A
            2 6 3   1    1 2    2 3    2
            Oh, the bells, bells, bells!
            []| [-] [---]| [---]| [---]|

            A    A D    D     D      E
            4    6 4    5     6      1
            What a tale their terror tells
            [--] | [--] [---] [----] [---]

                  B         A
                  3     E   3
                  Of Despair!
                  [] [-----]|

       A             F B        F B   A   A
       3      D 2    3 1   3    4 2   1   4
       How they clang, and clash, and roar!
       [-] [--] [---]| [-] [---]| [-] [--]|

       A    B A
       5    3 4         E    F
       What a horror they outpour
       [--] | [----] [--] [-----]

B  E         B  E               B  A
4  2       3 5  3   1           4  5
On the bosom of the palpitating air!
[] [-] [---] [] [-] [---------] [-]|

           E   A   A             F
       F   4   1   1      1 E    5
       Yet the ear it fully knows,
       [-] [-] [-] [] [---] [---]|

               E           F
             2 5       1   6
            By the twanging,
            [] [-] [------]|

            B   E
            5   6   4
            And the clanging,
            [-] [-] [------]

         A   F          A    B   A
         5   1      2   2    6   3    A
         How the danger ebbs and flows;
         [-] [-] [----] [--] [-] [---]|

           F   A              F
       3   2   3         5    3
       Yet the ear distinctly tells,
       [-] [-] [-] [--------] [---]

            A  F
            2  4   A
            In the jangling,
            [] [-] [------]

            C   F
            1   5       3    E
            And the wrangling.
            [-] [-] [-------]|

       A   F                C   A
       6   6      4      F  2   3
       How the danger sinks and swells,
       [-] [-] [----] [---] [-] [----]

    B          B   B           A   B        C   B  A
 4  1     1    6   2         5 3   3    6   1   4  4    F
By the sinking or the swelling in the anger of the bells—
[] [-] [-----] [] [-] [------] [] [-] [---] [] [-] [---]|

             C   B  A
             2   5  5    1
             Of the bells—
             [] [-] [---]|

     C   B  A      B      B      B
     3   6  6      1      2      3
     Of the bells, bells, bells, bells,
     [] [-] [---]  [---]  [---]  [---]

            B      B      B
            4      5      6    2
            Bells, bells, bells—
            [---]  [---]  [---]|

 A   C         C    C  A       C   C  C    A
 4   1  6      3    2  1       4   3  1    6
 In the clamor and the clangor of the bells!
 [] [-] [----] [-] [-] [-----] [] [-] [---]|

 CF
IV.
[]|

          C     C   C      C   C  C
          4     5   5      6   6  2    3
          Hear the tolling of the bells—
          [--] [-] [-----] [] [-] [---]|

                 A    C    B
                 5    3    1
                 Iron bells!
                 [--] [---]|

A    C B     D            A     D                  B
6    4 1     1      4     4     1    5        6    2
What a world of solemn thought their monody compels!
[--] | [---] [] [----] [-----] [---] [----] [-----]|

        A   D  A       D   D  A
        6   2  4       2   3  3
        In the silence of the night,
        [] [-] [-----] [] [-] [---]

        D   B         B    C
        4   2     D   3    5
        How we shiver with affright
        [-] [] [----] [--] [------]

  C   D             A      D   D     D  B
  6   5           5 1      3   6     4  3
  At the melancholy menace of their tone!
  [] [-] [--------] [----] [] [---] [--]|

        A           A    E   A
        4    E      5    1   5
        For every sound that floats
        [-] [---] [---] [--] [----]

        A     E   A   B       E     E
        6     2   6   4       3     4
        From the rust within their throats
        [--] [-] [--] [----] [---] [-----]

                 B  D A
                 1  1 1    1
                 Is a groan.
                 [] | [---]|

        D    E         D    E
        2    5  2     43    6  3     5
        And the people—ah, the people—
        [-] [-] [----]|[]  [-] [----]|

             F   A        B   F
          6  1   1      4 2   2      5
       They that dwell up in the steeple,
       [--] [--] [---] [] [] [-] [-----]

                 D   D
                 4   5
                 All alone,
                 [-] [---]

        D   B    D        D        E
        6   5    5        6        1
        And who tolling, tolling, tolling,
        [-] [-] [-----]  [-----]  [-----]

          B   F    B      A
          3   3    1      2
          In that muffled monotone,
          [] [--] [-----] [------]

         B    E A     B  A  A
         1    1 2     4  5  2
         Feel a glory in so rolling
         [--] | [---] [] [] [-----]

          E   F   B    F     E A
          2   4   2    5     2 6    6
          On the human heart a stone—
          [] [-] [---] [---] | [---]|

      F   E   A       A   A   B    A
      6   3   4       3   5   6    1
     They are neither man nor woman—
     [--] [-] [-----] [-] [-] [---]|

       A  E   A         B   B    B   A
       4  4   6         3   1    4   2
     They are neither brute nor human—
     [--] [-] [-----] [---] [-] [---]|

                A  E      B
                5  5      5  _
              They are Ghouls:
              [--] [-] [----]|

        E     A        B  B  C    E
        6     6   2    5  6  1    3   B
        And their king it is who tolls;
        [-] [---] [--] [] [] [-] [---]|

        F    B A      A      A
        1    1 3      4      5
        And he rolls, rolls, rolls,
        [-] [] [---]  [---]  [---]

                    A
                    6
                    Rolls
                    [---]

             F      B      B C    B
             2 6    2      2 4    4
             A pæan from the bells!
             | [--] [--] [-] [---]|

          F    C  A     C     B
          3    1  4     5     1
          And his merry bosom swells
          [-] [-] [---] [---] [----]

             C      B A    E    B C    B
             2      3 1    4    4 6    5
             With the pæan of the bells!
             [--] [-] [--] [] [-] [---]|

          F    B A       F    B A
          4    5 2       5    6 1    C
          And he dances, and he yells;
          [-] [] [----]  [-] [] [---]|

                   C     C     C
          3        2     3     4
          Keeping time, time, time,
          [-----] [--]  [--]  [--]

          C  F B    E   B    B
          5  6 2    5   6    1
          In a sort of Runic rhyme,
          [] | [--] [] [---] [---]

              E   C A    F    C D    A
              6   1 2    1    2 1    3
             To the pæan of the bells—
             [] [-] [--] [] [-] [---]|

               F    C D
               2    3 2    A
               Of the bells:
               [] [-] [---]|

                   C     D     D
          4        6     1     2
          Keeping time, time, time,
          [-----] [--]  [--]  [--]

          D    B    F   C    B
          3    3    3   1    2
          In a sort of Runic rhyme,
          []   [--] [] [---] [---]

             F   C   B       F    C D    A
             4   4   3       5    5 3    4
            To the throbbing of the bells—
            [] [-] [-------] [] [-] [---]|

          F    C D      D      D    A
          6    6 4      5      6    5
          Of the bells, bells, bells—
          [] [-] [---]  [---]  [---]|

                 D B        B   D E
                 1 4        3   2 1    D
            To the sobbing of the bells;
               [-] [-----] [] [-] [---]|

                   D     D     D
          5        4     5     6
          Keeping time, time, time,
          [-----] [--]  [--]  [--]

             B  D         A       A
             5  3 6       1       2
            As he knells, knells, knells,
            [] [] [----]  [----]  [----]

          E      A    C    B
          1      3    2    4
          In a happy Runic rhyme,
          []   [---] [---] [---]

                 D B        B   D E    A
                 4 5        4   5 2    6
            To the rolling of the bells—
               [-] [-----] [] [-] [---]|

           B   D E      E      E    B
           5   6 3      4      5    1
          Of the bells, bells, bells—
          [] [-] [---]  [---]  [---]|

                 E   A      B   E E
                 1   1      6   2 6
            To the tolling of the bells,
               [-] [-----] [] [-] [---]

       C   E F      F      F      F    B
       1   3 1      2      3      4    2
      Of the bells, bells, bells, bells—
      [] [-] [---]  [---]  [---]  [---]|

              F      F       E   B
              5      6       4   3
              Bells, bells, bells—
              [---]  [---]  [---]|

       E A        B    E A         C   F  F
       5 5        2    6 3         2   1  2   2
  To the moaning and the groaning of the bells.
     [-] [-----] [-] [-] [------] [] [-] [---]|
